### DISCUSSION TITLE
Question mocking a react component for react-testing-library

### DISCUSSION THREAD

**Discussion by DoctaCloak** [vitest-dev/vitest#2900] [Q&A]
There doesn’t seem to be anything anywhere that discusses how to mock a component out in your test.

Back when we used jest I could do something like this but in vitest that doesn’t seem to be the case…

`jest.mock(‘pathToFile’, () => () => <div>hello world</div>)`

This is the only thing preventing me from suggesting the lib to my team and I really want to…please help.

**Comment by sheremet-va** [comment]
The migration guide mentions mocking default exports: https://vitest.dev/guide/migration.html

In ESM module can only be an object, so you need to return an object with the default key:

```js
vi.mock('path', () => ({
  default: () => <div>hello world</div>
})
```

**Reply by DoctaCloak** [reply]
Ah you know what, I wasn’t returning a function in default!

**Reply by siyi98** [reply]
you saved me!
